modules/alloydb/main.tf

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+resource "random_pet" "alloydb" {
+  length = 2
+}
+
+resource "google_alloydb_instance" "tfe" {
+  cluster           = google_alloydb_cluster.tfe.name
+  instance_id       = "${var.namespace}-tfe-${random_pet.alloydb.id}-1"
+  instance_type     = "PRIMARY"
+  availability_type = var.availability_type
+  labels            = var.labels
+
+  machine_config {
+    cpu_count = 2
+  }
+  # You get a NETWORK_PEERING_DELETED error if you don't wait for the service networking connection to be created.
+  depends_on = [var.service_networking_connection]
+}
+
+resource "google_alloydb_cluster" "tfe" {
+  cluster_id = "${var.namespace}-tfe-${random_pet.alloydb.id}"
+
+  database_version = var.postgres_version
+  location         = "us-east4"
```

Comment:
Will the location ever need to change or does it need to match the instance location? If not, then it's okay to hard code.